hack/deploy-cert-manager.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+set -eo pipefail
+
+OCP_MINOR=$(oc get clusterversion version -o jsonpath='{.status.desired.version}' | cut -d. -f2)
+echo "Detected OCP minor version: ${OCP_MINOR}"
+
+TARGET=$((OCP_MINOR - 3))
+echo "Target cert-manager minor version: ${TARGET}"
+
+# Get package manifest JSON
+PACKAGE_MANIFEST=$(oc get packagemanifest openshift-cert-manager-operator -n openshift-marketplace -o json)
+
+# Find version-specific channels matching pattern stable-vX.Y
+COMPATIBLE_CHANNELS=$(echo "$PACKAGE_MANIFEST" | jq -r --arg target "$TARGET" '
+  .status.channels[] | 
+  select(.name | test("stable-v[0-9]+\\.[0-9]+$")) |
+  .name as $name |
+  ($name | split("-v")[1] | split(".") | map(tonumber)) as $version |
+  {
+    name: $name,
+    major: $version[0],
+    minor: $version[1],
+    currentCSV: .currentCSV
+  } |
+  select(.minor <= ($target | tonumber))
+')
+
+SELECTED_CHANNEL=$(echo "$COMPATIBLE_CHANNELS" | jq -s 'sort_by(.minor) | last')
+
+# Fallback to default channel if no version-specific match
+if [[ -z "$SELECTED_CHANNEL" || "$SELECTED_CHANNEL" == "null" ]]; then
+  echo "No version-specific channel found, using default channel"
+  DEFAULT_CHANNEL=$(echo "$PACKAGE_MANIFEST" | jq -r '.status.defaultChannel')
+  SELECTED_CHANNEL=$(echo "$PACKAGE_MANIFEST" | jq -r --arg ch "$DEFAULT_CHANNEL" '
+    .status.channels[] | select(.name == $ch) | {
+      name: .name,
+      currentCSV: .currentCSV
+    }
+  ')
+fi
+
+CHANNEL_NAME=$(echo "$SELECTED_CHANNEL" | jq -r '.name')
+STARTING_CSV=$(echo "$SELECTED_CHANNEL" | jq -r '.currentCSV')
+
+echo "Selected channel: ${CHANNEL_NAME}"
+echo "Starting CSV: ${STARTING_CSV}"
+
+# Deploy using template
+envsubst <<EOF | oc apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cert-manager-operator
+  labels:
+    name: cert-manager-operator
+---
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: openshift-cert-manager-operator
+  namespace: cert-manager-operator
+spec:
+  targetNamespaces:
```

Comment:
https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/security_and_compliance/cert-manager-operator-for-red-hat-openshift#cert-manager-install-cli_cert-manager-operator-install

This may get more challenging to support than we think.

Reply:
Tweaked this a bit to address the concern for pre 1.15 releases.